design/one-pager-trait-workload-interaction-mechanism.md

```diff
@@ -122,9 +122,23 @@ The overall idea is for the applicationConfiguration controller to fill critical
 in the workload and trait CR it emits. In addition, we will provide a helper library so that
 trait controller developers can locate the resources they need with a simple function call. 
 Here is the list of changes that we propose.
-1. ApplicationConfig controller no longer assumes that all `trait` CRDs contain a "spec
+1. Add an optional field called `oamAware` to the `traitDefinition` schema. The trait owner needs
```

Comment:
The name `oamAware` is not intuitive, what about "workloadDefNeeded"

Reply:
I am open to naming. However, the reason for this name is we may reuse this field for future purposes, not limited to workload. So this is intended to be a general-purpose field.

Reply:
Yeah, this is a bad name.

Actually, I would suggest change `workloadRef` to `targetRef`, and change this flag to `TargetAware`.

Note that `targetRef` also aligns with what Kubernetes is using today.

Reply:
@resouer can you point me to where `targetRef` is used in K8s?

Reply:
Check this one: https://cloud.google.com/kubernetes-engine/docs/concepts/verticalpodautoscaler k8s has a built-in type for it.

Reply:
@ryanzhang-oss I will eat my word on the field name. `targetRef` is overlapping. Let's keep `workloadRef`.